### PR TITLE
docs(karma): add running locally documentation

### PR DESCRIPTION
**Pull request by rwaskiewicz**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [ ] Build (`npm run build`) was run locally and any changes were pushed
- [ ] Unit tests (`npm test`) were run locally and passed
- [ ] E2E Tests (`npm run test.karma.prod`) were run locally and passed
- [ ] Prettier (`npm run prettier`) was run locally and passed

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [ ] Feature
- [x] Refactoring (no functional changes, no api changes)
- [x] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Previously, testing a change that affected our Karma/Browserstack tests involved one of two things:
- Creating a Draft PR and continuously pushing to a WIP branch to test changes (which wastes time waiting for builds to pass/complete)
- Hack at the Karma configuration to tunnel into Browserstack locally

GitHub Issue Number: N/A

## What is the new behavior?

this commit documents the ability to run the karma tests locally. in
addition, it also removes hacks that developers once had to do in order
to connect to browserstack locally, by introducing a new environment
variable, LOCAL_BROWSERSTACK, that will allow the development team to
tunnel into browserstack for local testing purposes

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Testing

Run through the README changes, validated that the new env var works as expected

## Other information

This is being done as a part of STENCIL-433: Stabilize Browserstack CI Tests

### CODE REVIEW

File: test/karma/README.md [hunk 1]
```diff
@@ -1,15 +1,57 @@
...
+
+To tunnel into Browserstack, an access key and username are required.
+These can be accessed through the [Browserstack Dashboard](https://automate.browserstack.com/dashboard/v2).
+At this time, only members of the Stencil core team are granted access keys/usernames.
+If you are a member of the Stencil team and do not have access to Browserstack, please see the team lead.
+
+Once an access key and username have been acquired, tunneling into Browserstack requires invoking the tests with the
+following environment variables:
+```
+LOCAL_BROWSERSTACK=1 BROWSERSTACK_ACCESS_KEY=[KEY] BROWSERSTACK_USERNAME=[USERNAME] npm run karma
```
Comment (alicewriteswrongs) [inline test/karma/README.md:40]: should this command be `npm run test.karma.prod` ?

**Review by alicewriteswrongs — CHANGES_REQUESTED** [review]
code changes look good, and I'm able to run `npm run test.karma.prod` locally w/ a browserstack connection (getting some test failures though!) but just noticed one thing in the README

File: test/karma/README.md (hunk 1 above)
Comment (rwaskiewicz) [reply test/karma/README.md:40]: Hmm actually that's a good question. It depends on the context - I had done my building and was in the `test/karma` directory when I was running this.  I'll add to the documentation to clarify

Comment (alicewriteswrongs) [reply test/karma/README.md:40]: ahh I see, that makes sense

**Review by alicewriteswrongs — APPROVED** [review]
lgtm! 👍

**Pull request merged**